fair_svm/fair_svm.py

```diff
@@ -0,0 +1,356 @@
+        SVC. If not given, all classes are supposed to have
+        weight one.
+        The "balanced" mode uses the values of y to automatically adjust
+        weights inversely proportional to class frequencies in the input data
+        as ``n_samples / (n_classes * np.bincount(y))``
+
+    max_iter : int, optional (default=-1)
+        Hard limit on iterations within solver, or -1 for no limit.
+
+    random_state : int seed, RandomState instance, or None (default)
+        The seed of the pseudo random number generator to use when
+        shuffling the data for probability estimation.
+
+    Attributes
+    ----------
+    support_ : array-like, shape = [n_SV]
+        Indices of support vectors.
+
+    support_vectors_ : array-like, shape = [n_SV, n_features]
+        Support vectors.
+
+    dual_coef_ : array, shape = [n_class-1, n_SV]
+        Coefficients of the support vector in the decision function.
+        For multiclass, coefficient for all 1-vs-1 classifiers.
+        The layout of the coefficients in the multiclass case is somewhat
+        non-trivial. See the section about multi-class classification in the
+        SVM section of the User Guide for details.
+
+    coef_ : array, shape = [n_class-1, n_features]
+        Weights assigned to the features (coefficients in the primal
+        problem). This is only available in the case of a linear kernel.
+
+        `coef_` is a readonly property derived from `dual_coef_` and
+        `support_vectors_`.
+
+    intercept_ : array, shape = [n_class * (n_class-1) / 2]
+        Constants in decision function.
+
+
+    """
+
+    def __init__(self, C=1.0, kernel='rbf', degree=3, gamma='auto',
+                 coef0=0.0, 
+                 tol=1e-3,  class_weight=None,
+                 max_iter=1000, 
+                 random_state=None):
+
+        self.C = C
+        self.kernel = kernel
+        self.degree = degree
+        self.gamma = gamma
+        self.coef0 = coef0
+        self.tol = tol
+        self.class_weight = class_weight
+        self.max_iter = max_iter
+        self.random_state = random_state
+
+        np.random.seed(self.random_state)
+
+        self.gram_matrix = None # gram matrix will be computed in the fit method
+        
+
+
+    def compute_and_save_kernel(self, X1, X2):
+
+       # # if the kernel is a callable, then compute it
+        if callable(self.kernel):            
+            gram_matrix = self.kernel(X1, X2)
+            return gram_matrix
+
+
+        # TODO: Check all the kernels again
+        ## Compute the kernel
+        if self.kernel == "linear":
+            gram_matrix = linear_kernel(X1, X2)
+        elif self.kernel == "poly":
+            gram_matrix = polynomial_kernel(X1, X2, degree = self.degree, gamma=self.gamma, coef0 = self.coef0)
+        elif self.kernel == "rbf":
+            gram_matrix = rbf_kernel(X1, X2, gamma = self.gamma)
+        elif self.kernel == "sigmoid":
```

Comment:
Maybe we should follow this with an else case, which raises an exception if the kernel isn't any of the expected values.